src/forms-opportunities/utils.js

```diff
@@ -88,16 +91,53 @@ function aggregateFormVitalsByDevice(formVitalsCollection) {
   return resultMap;
 }
 
-function convertToOpportunityData(opportunityName, urlObject) {
+export async function generatePresignedUrls(screenshots, s3Key, s3ClientObj, log) {
+  return Promise.all(screenshots.map(async (screenshot) => {
+    const screenshotPath = `${s3Key}${screenshot.fileName}`;
+    log.info(`debug log screenshot path ${screenshotPath}`);
+    // eslint-disable-next-line max-len
+    const command = new GetObjectCommand({ Bucket: process.env.S3_BUCKET_NAME, Key: screenshotPath });
+
+    return getSignedUrl(s3ClientObj, command, { expiresIn: EXPIRY_IN_SECONDS })
+      .then((presignedUrl) => ({ ...screenshot, presignedurl: presignedUrl }))
+      .catch((error) => {
+        log.error(`Error generating presigned URL for ${screenshot.fileName}: ${error.message}`);
+        return { ...screenshot, presignedurl: '' };
+      });
+  }));
+}
+
+async function convertToOpportunityData(opportunityName, urlObject, scrapedData, context) {
   const {
     url, pageViews, formViews, formSubmit, CTA,
   } = urlObject;
+
+  const {
+    log, s3Client: s3ClientObj,
+  } = context;
+
+  // Find matching entry in scrapedData
+  let screenshots = [];
+  let s3Key = '';
+
   let conversionRate = formSubmit / formViews;
   conversionRate = Number.isNaN(conversionRate) ? null : conversionRate;
 
+  if (scrapedData) {
```

Comment:
`  we can just do it like: const screenshot = await getPresignedUrl('screenshot-desktop.png', context, formUrl, site);`